src/Canvas.tsx

```diff
@@ -113,33 +105,34 @@ function Canvas<R, K extends keyof R>({
     return height - scrollbarSize;
   }
 
+  function getFrozenColumnsWidth() {
+    if (lastFrozenColumnIndex === -1) return 0;
+    const lastFrozenCol = columns[lastFrozenColumnIndex];
+    return lastFrozenCol.left + lastFrozenCol.width;
+  }
+
   function scrollToCell({ idx, rowIdx }: Partial<Position>) {
     const { current } = canvas;
     if (!current) return;
 
+    const { clientWidth, clientHeight } = current;
+
     if (typeof idx === 'number' && idx > lastFrozenColumnIndex) {
       const { left, width } = columns[idx];
-      const { clientWidth, scrollLeft } = current;
-      const isCellAtLeftBoundary = left < scrollLeft + width + frozenColumnsWidth;
+      const isCellAtLeftBoundary = left < scrollLeft + width + getFrozenColumnsWidth();
       const isCellAtRightBoundary = left + width > clientWidth + scrollLeft;
       if (isCellAtLeftBoundary || isCellAtRightBoundary) {
         const newScrollLeft = getColumnScrollPosition(columns, idx, scrollLeft, clientWidth);
-        if (newScrollLeft !== 0) {
-          current.scrollLeft = scrollLeft + newScrollLeft;
-        }
+        current.scrollLeft = scrollLeft + newScrollLeft;
```

Comment:
I ended up using `scrollLeft` and `scrollTop` props. They should have the same value I believe